Classes/TYPO3/TYPO3CR/Search/Eel/IndexingHelper.php

```diff
@@ -102,12 +122,61 @@ public function convertArrayOfNodesToArrayOfNodeIdentifiers($nodes) {
 		return $nodeIdentifiers;
 	}
 
+
+	/**
+	 * @param AssetInterface $value
+	 * @return array
+	 */
+	public function extractAsset($value) {
+
+		$asset = array();
+		if($value !== NULL) {
+			$resource = $value->getResource();
+			$asset['assetIdentifiers'] = $this->persistenceManager->getIdentifierByObject($resource);
+			$asset['assetPaths'] = $this->resourcePublisher->getPersistentResourceWebUri($resource);
+			$asset['assetFilenames'] = $resource->getFilename();
+			$asset['assetFileExtensions'] = $resource->getFileExtension();
+		}
+		return $asset;
+	}
+
+	/**
+	 * @param array<TYPO3\Media\Domain\Model\AssetInterface> $value
+	 * @return array
+	 */
+	public function extractAssetList($value) {
+
+		$assetList = array(
+			'assetIdentifiers' => '',
+			'assetPaths' => '',
+			'assetFilenames' => '',
+			'assetFileExtensions' => ''
+		);
+
+		foreach($value as $asset) {
+			$asset = $this->extractAsset($asset);
+			$assetList['assetIdentifiers'] .= $asset['assetIdentifiers'] . ', ';
+			$assetList['assetPaths'] .= $asset['assetPaths'] . ', ';
+			$assetList['assetFilenames'] .= $asset['assetFilenames'] . ', ';
+			$assetList['assetFileExtensions'] .= $asset['assetFileExtensions'] . ', ';
+		}
+
+		$assetList['assetIdentifiers'] = trim($assetList['assetIdentifiers'], ', ');
+		$assetList['assetPaths'] = trim($assetList['assetPaths'], ', ');
+		$assetList['assetFilenames'] = trim($assetList['assetFilenames'], ', ');
+		$assetList['assetFileExtensions'] = trim($assetList['assetFileExtensions'], ', ');
+
+		return $assetList;
+	}
+
 	/**
 	 *
 	 * @param $string
+	 * @param $bucket
 	 * @return array
 	 */
-	public function extractHtmlTags($string) {
+	public function extractHtmlTags($string, $bucket = NULL) {
```

Comment:
I have to investigate again. Long time ago. But I think otherwise only h1-6 and p tags were able to be fulltext indexed. Will rebuild an according setup, so we can discuss this topic at the codesprint